docs/semgrep-secrets/historical-scanning.md

```diff
@@ -0,0 +1,120 @@
+
+You can perform one-time historical scans or enable historical scanning for all Secrets scans. Detecting valid secrets in your git history is a first step to reducing your repository's attack surface.
+
+You can run historical scans in the CLI and in your Semgrep deployment, which enables you to track and triage these secrets.
+
+:::note Feature maturity
+- This feature is in **public beta**. See [Limitations](#limitations) for more information.
+- All Semgrep Secrets customers can enable this feature.
+- Please leave feedback by either reaching out to your technical account manager (TAM) or through the **<i class="fa-solid fa-bullhorn"></i> Feedback** form in Semgrep Cloud Platform's navigation bar.
+:::
+
+## Run historical scans
+
+You can enable historical scanning for all of your future secret scans or run a dedicated CI job for one-time scans. By default, historical scans display **valid, leaked secrets** to ensure a high true positive rate.
+
+### Prerequisites
+
+- **CLI tool**: Historical scanning requires at least Semgrep **v1.65.0**. See [Update](/update/) for instructions.
+- **Single-tenant Semgrep Cloud Platform (SCP)**: Reach out to your TAM to ensure your instance is up-to-date.
+
+### Enable historical scanning for all Secrets scans
+
+:::tip
+If possible, [test historical scans locally](#run-a-local-test-scan) to create a benchmark of performance and scan times before adding historical scans to your formal security process.
+:::
+
+1. Sign in to Semgrep Cloud Platform.
+1. Click **<i class="fa-solid fa-gear"></i> Settings**.
+1. Under Deployment, click the **<i class="fa-solid fa-toggle-large-on"></i> Historical scanning** toggle.
+![Historical scanning settings toggle](/img/historical-scanning-settings.png#md-width)
+
+Subsequent Semgrep full scans now include historical scanning.
+
+### Run a one-off historical scan
+
+To run a one-off or on-demand historical scan, you can create a specific CI job and then manually start the job as needed.
+
+The general steps are:
+
+1. Copy your current full scan CI job configuration file.
+1. Look for the `semgrep ci` command.
+1. Append the `--historical-secrets` flag:
+    ```
+    semgrep ci --historical-secrets
+    ```
+1. Depending on your CI provider, you may have to perform additional steps to enable the job to run manually. For example, GitHub Actions requires the `workflow_dispatch` event to be added to your CI job.
+
+
+### Run a local test scan
+
+You can run a historical scan locally without sending the scan results to Semgrep Cloud Platform.
+
+You can run this test to determine the time it takes for Semgrep Secrets to run on your repository's git commit history.
+
+To run a test scan, enter the following command:
+
+```bash
+semgrep scan --historical-secrets
+```
+
+The historical scan results appear in the **Secrets Historical Scan** section:
+
+![Historical scan section in the CLI](/img/historical-scans-cli.png#md-width)
+
+#### Find all secrets
```

Comment:
I don't think we want to publicly acknowledge this "feature" (more like unintended feature) since it might go away in the future.